{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# run to export data from the database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "To customize the output files please enter these credentials: \n",
      "Enter your name: Danilo\n",
      "Enter your surname: Pignotti\n",
      "enter your username: danilo\n",
      "enter your password: ········\n",
      "\n",
      "tables selection menu from which to export data:\n",
      "\n",
      "enter 1 to export data from database MIQUALAT following table: GENE;\n",
      "enter 2 to export data from database MIQUALAT following table: DB;\n",
      "enter 3 to export data from database MIQUALAT following table: GEN_DB;\n",
      "enter 4 to export data from database MIQUALAT following table: PUBLICATION;\n",
      "enter 5 to export data from database MIQUALAT following table: TECNIQUE;\n",
      "enter 6 to export data from database MIQUALAT following table: TAG;\n",
      "enter 7 to export data from database MIQUALAT following table: VARIANT;\n",
      "enter 8 to export data from database MIQUALAT following table: PUB_GEN_VAR_TEC_TAG;\n",
      "\n",
      "enter tables selection numbers (i.e. 7 or 1,2,3): 1,2,3,4,5,6,7,8\n",
      "\n",
      "queires selection menu to export data from table GENE:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract specific record relative to an input ensembl ID; \n",
      "3: to extract specific record relative to an input gene_name; \n",
      "4: to extract all the species and relative refseq version; \n",
      "5: to count genes total numbers sorted by species; \n",
      "6: to extract all genes from an input species; \n",
      "7: to extract all genes located on an input chromosome for an input species; \n",
      "8: to extract all genes located between start and end input coordinates on an input chromosome for an input species;\n",
      "\n",
      "enter GENE table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_ID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_short_description</th>\n",
       "      <th>refseq</th>\n",
       "      <th>species</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_coordinate</th>\n",
       "      <th>end_coordinate</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSBTAG00000000005</td>\n",
       "      <td>GRK3</td>\n",
       "      <td>G protein-coupled receptor kinase 3</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>17</td>\n",
       "      <td>65389743</td>\n",
       "      <td>65505336</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSBTAG00000000008</td>\n",
       "      <td>KCNJ1</td>\n",
       "      <td>potassium inwardly rectifying channel subfamil...</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>29</td>\n",
       "      <td>32214439</td>\n",
       "      <td>32244810</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSBTAG00000000009</td>\n",
       "      <td>FOXF1</td>\n",
       "      <td>forkhead box F1</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>18</td>\n",
       "      <td>12338037</td>\n",
       "      <td>12342272</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSBTAG00000000010</td>\n",
       "      <td>UBL7</td>\n",
       "      <td>ubiquitin like 7</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>21</td>\n",
       "      <td>34209956</td>\n",
       "      <td>34223394</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSBTAG00000000011</td>\n",
       "      <td>TDH</td>\n",
       "      <td>L-threonine dehydrogenase</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>8</td>\n",
       "      <td>7950815</td>\n",
       "      <td>7971600</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27265</th>\n",
       "      <td>ENSBTAG00000055312</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>2</td>\n",
       "      <td>96551552</td>\n",
       "      <td>96557130</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27266</th>\n",
       "      <td>ENSBTAG00000055313</td>\n",
       "      <td>bta-mir-12025</td>\n",
       "      <td>bta-mir-12025</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>21</td>\n",
       "      <td>65169462</td>\n",
       "      <td>65169520</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27267</th>\n",
       "      <td>ENSBTAG00000055314</td>\n",
       "      <td>OR3A1</td>\n",
       "      <td>olfactory receptor, family 3, subfamily A, mem...</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>19</td>\n",
       "      <td>24021135</td>\n",
       "      <td>24051219</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27268</th>\n",
       "      <td>ENSBTAG00000055315</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>8</td>\n",
       "      <td>102574538</td>\n",
       "      <td>102575815</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27269</th>\n",
       "      <td>ENSBTAG00000055316</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>2</td>\n",
       "      <td>94738416</td>\n",
       "      <td>94738887</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27270 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          ensembl_gene_ID      gene_name  \\\n",
       "0      ENSBTAG00000000005           GRK3   \n",
       "1      ENSBTAG00000000008          KCNJ1   \n",
       "2      ENSBTAG00000000009          FOXF1   \n",
       "3      ENSBTAG00000000010           UBL7   \n",
       "4      ENSBTAG00000000011            TDH   \n",
       "...                   ...            ...   \n",
       "27265  ENSBTAG00000055312           None   \n",
       "27266  ENSBTAG00000055313  bta-mir-12025   \n",
       "27267  ENSBTAG00000055314          OR3A1   \n",
       "27268  ENSBTAG00000055315           None   \n",
       "27269  ENSBTAG00000055316           None   \n",
       "\n",
       "                                  gene_short_description      refseq  \\\n",
       "0                   G protein-coupled receptor kinase 3   ARS-UCD1.2   \n",
       "1      potassium inwardly rectifying channel subfamil...  ARS-UCD1.2   \n",
       "2                                       forkhead box F1   ARS-UCD1.2   \n",
       "3                                      ubiquitin like 7   ARS-UCD1.2   \n",
       "4                             L-threonine dehydrogenase   ARS-UCD1.2   \n",
       "...                                                  ...         ...   \n",
       "27265                                               None  ARS-UCD1.2   \n",
       "27266                                     bta-mir-12025   ARS-UCD1.2   \n",
       "27267  olfactory receptor, family 3, subfamily A, mem...  ARS-UCD1.2   \n",
       "27268                                               None  ARS-UCD1.2   \n",
       "27269                                               None  ARS-UCD1.2   \n",
       "\n",
       "          species  chromosome  start_coordinate  end_coordinate  strand  \n",
       "0      bos taurus          17          65389743        65505336       1  \n",
       "1      bos taurus          29          32214439        32244810      -1  \n",
       "2      bos taurus          18          12338037        12342272       1  \n",
       "3      bos taurus          21          34209956        34223394       1  \n",
       "4      bos taurus           8           7950815         7971600      -1  \n",
       "...           ...         ...               ...             ...     ...  \n",
       "27265  bos taurus           2          96551552        96557130       1  \n",
       "27266  bos taurus          21          65169462        65169520       1  \n",
       "27267  bos taurus          19          24021135        24051219      -1  \n",
       "27268  bos taurus           8         102574538       102575815       1  \n",
       "27269  bos taurus           2          94738416        94738887       1  \n",
       "\n",
       "[27270 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table DB:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract all record fields relative to an input database_ID; \n",
      "3: to extract all record fields relative to an input compound name (or pathway or molecule) to search into field database_object_name;\n",
      "\n",
      "enter DB table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>database_ID</th>\n",
       "      <th>database_object_type</th>\n",
       "      <th>database_object_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GO:0000166</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>nucleotide binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GO:0000287</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>magnesium ion binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GO:0004601</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>peroxidase activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GO:0004602</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>glutathione peroxidase activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GO:0005391</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>sodium:potassium-exchanging ATPase activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>GO:0005524</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>ATP binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>GO:0005783</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>endoplasmic reticulum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GO:0005802</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>trans-Golgi network</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GO:0005886</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>plasma membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>GO:0006811</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>ion transport</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GO:0006813</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>potassium ion transport</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GO:0006883</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>cellular sodium ion homeostasis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GO:0006979</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>response to oxidative stress</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>GO:0008556</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>potassium transmembrane transporter activity, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>GO:0008900</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>potassium:proton exchanging ATPase activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>GO:0010248</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>establishment or maintenance of transmembrane ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GO:0015914</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>phospholipid transport</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>GO:0016020</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>GO:0016021</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>integral component of membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>GO:0016491</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>oxidoreductase activity</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>GO:0019829</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>ATPase-coupled cation transmembrane transporte...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>GO:0030007</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>cellular potassium ion homeostasis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>GO:0036376</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>sodium ion export across plasma membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>GO:0045332</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>phospholipid translocation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>GO:0046872</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>metal ion binding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>GO:0055037</td>\n",
       "      <td>cellular_component</td>\n",
       "      <td>recycling endosome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>GO:0055114</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>oxidation-reduction process</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>GO:0071805</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>potassium ion transmembrane transport</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>GO:0098655</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>cation transmembrane transport</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>GO:0098869</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>cellular oxidant detoxification</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>GO:0140326</td>\n",
       "      <td>molecular_function</td>\n",
       "      <td>ATPase-coupled intramembrane lipid transporter...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>GO:1902600</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>proton transmembrane transport</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>GO:1990573</td>\n",
       "      <td>biological_process</td>\n",
       "      <td>potassium ion import across plasma membrane</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>R-BTA-3299685</td>\n",
       "      <td>reactome_pathway</td>\n",
       "      <td>Detoxification of Reactive Oxygen Species</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>R-BTA-936837</td>\n",
       "      <td>reactome_pathway</td>\n",
       "      <td>Ion transport by P-type ATPases</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      database_ID database_object_type  \\\n",
       "0      GO:0000166   molecular_function   \n",
       "1      GO:0000287   molecular_function   \n",
       "2      GO:0004601   molecular_function   \n",
       "3      GO:0004602   molecular_function   \n",
       "4      GO:0005391   molecular_function   \n",
       "5      GO:0005524   molecular_function   \n",
       "6      GO:0005783   cellular_component   \n",
       "7      GO:0005802   cellular_component   \n",
       "8      GO:0005886   cellular_component   \n",
       "9      GO:0006811   biological_process   \n",
       "10     GO:0006813   biological_process   \n",
       "11     GO:0006883   biological_process   \n",
       "12     GO:0006979   biological_process   \n",
       "13     GO:0008556   molecular_function   \n",
       "14     GO:0008900   molecular_function   \n",
       "15     GO:0010248   biological_process   \n",
       "16     GO:0015914   biological_process   \n",
       "17     GO:0016020   cellular_component   \n",
       "18     GO:0016021   cellular_component   \n",
       "19     GO:0016491   molecular_function   \n",
       "20     GO:0019829   molecular_function   \n",
       "21     GO:0030007   biological_process   \n",
       "22     GO:0036376   biological_process   \n",
       "23     GO:0045332   biological_process   \n",
       "24     GO:0046872   molecular_function   \n",
       "25     GO:0055037   cellular_component   \n",
       "26     GO:0055114   biological_process   \n",
       "27     GO:0071805   biological_process   \n",
       "28     GO:0098655   biological_process   \n",
       "29     GO:0098869   biological_process   \n",
       "30     GO:0140326   molecular_function   \n",
       "31     GO:1902600   biological_process   \n",
       "32     GO:1990573   biological_process   \n",
       "33  R-BTA-3299685     reactome_pathway   \n",
       "34   R-BTA-936837     reactome_pathway   \n",
       "\n",
       "                                 database_object_name  \n",
       "0                                  nucleotide binding  \n",
       "1                               magnesium ion binding  \n",
       "2                                 peroxidase activity  \n",
       "3                     glutathione peroxidase activity  \n",
       "4         sodium:potassium-exchanging ATPase activity  \n",
       "5                                         ATP binding  \n",
       "6                               endoplasmic reticulum  \n",
       "7                                 trans-Golgi network  \n",
       "8                                     plasma membrane  \n",
       "9                                       ion transport  \n",
       "10                            potassium ion transport  \n",
       "11                    cellular sodium ion homeostasis  \n",
       "12                       response to oxidative stress  \n",
       "13  potassium transmembrane transporter activity, ...  \n",
       "14        potassium:proton exchanging ATPase activity  \n",
       "15  establishment or maintenance of transmembrane ...  \n",
       "16                             phospholipid transport  \n",
       "17                                           membrane  \n",
       "18                     integral component of membrane  \n",
       "19                            oxidoreductase activity  \n",
       "20  ATPase-coupled cation transmembrane transporte...  \n",
       "21                 cellular potassium ion homeostasis  \n",
       "22           sodium ion export across plasma membrane  \n",
       "23                         phospholipid translocation  \n",
       "24                                  metal ion binding  \n",
       "25                                 recycling endosome  \n",
       "26                        oxidation-reduction process  \n",
       "27              potassium ion transmembrane transport  \n",
       "28                     cation transmembrane transport  \n",
       "29                    cellular oxidant detoxification  \n",
       "30  ATPase-coupled intramembrane lipid transporter...  \n",
       "31                     proton transmembrane transport  \n",
       "32        potassium ion import across plasma membrane  \n",
       "33          Detoxification of Reactive Oxygen Species  \n",
       "34                    Ion transport by P-type ATPases  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table GEN_DB:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract database_id,ensembl_gene_ID,gene_name,species of all genes relative to an input compound name (or pathway or molecule) to search into field database_object_name; \n",
      "3: to extract all DB record fields related to an input species and an input database_object_type (gene, pathway, protein, enzyme, biological process, molecular function or others);\n",
      "\n",
      "enter GEN_DB table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_ID</th>\n",
       "      <th>database_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0000166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0000166</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0000287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>GO:0004601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>GO:0004602</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0005391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0005524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0005524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0005783</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0005802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0005886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0006811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0006813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0006883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>GO:0006979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0008556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0008900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0010248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0015914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0016020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0016020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0016021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0016021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>GO:0016491</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0019829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0030007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0036376</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0045332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0046872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0046872</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0055037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>GO:0055114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0071805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:0098655</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>GO:0098869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>GO:0140326</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:1902600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>GO:1990573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>ENSBTAG00000012023</td>\n",
       "      <td>R-BTA-3299685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>ENSBTAG00000014486</td>\n",
       "      <td>R-BTA-936837</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>ENSBTAG00000016484</td>\n",
       "      <td>R-BTA-936837</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ensembl_gene_ID    database_ID\n",
       "0   ENSBTAG00000014486     GO:0000166\n",
       "1   ENSBTAG00000016484     GO:0000166\n",
       "2   ENSBTAG00000016484     GO:0000287\n",
       "3   ENSBTAG00000012023     GO:0004601\n",
       "4   ENSBTAG00000012023     GO:0004602\n",
       "5   ENSBTAG00000014486     GO:0005391\n",
       "6   ENSBTAG00000014486     GO:0005524\n",
       "7   ENSBTAG00000016484     GO:0005524\n",
       "8   ENSBTAG00000016484     GO:0005783\n",
       "9   ENSBTAG00000016484     GO:0005802\n",
       "10  ENSBTAG00000016484     GO:0005886\n",
       "11  ENSBTAG00000014486     GO:0006811\n",
       "12  ENSBTAG00000014486     GO:0006813\n",
       "13  ENSBTAG00000014486     GO:0006883\n",
       "14  ENSBTAG00000012023     GO:0006979\n",
       "15  ENSBTAG00000014486     GO:0008556\n",
       "16  ENSBTAG00000014486     GO:0008900\n",
       "17  ENSBTAG00000014486     GO:0010248\n",
       "18  ENSBTAG00000016484     GO:0015914\n",
       "19  ENSBTAG00000014486     GO:0016020\n",
       "20  ENSBTAG00000016484     GO:0016020\n",
       "21  ENSBTAG00000014486     GO:0016021\n",
       "22  ENSBTAG00000016484     GO:0016021\n",
       "23  ENSBTAG00000012023     GO:0016491\n",
       "24  ENSBTAG00000014486     GO:0019829\n",
       "25  ENSBTAG00000014486     GO:0030007\n",
       "26  ENSBTAG00000014486     GO:0036376\n",
       "27  ENSBTAG00000016484     GO:0045332\n",
       "28  ENSBTAG00000014486     GO:0046872\n",
       "29  ENSBTAG00000016484     GO:0046872\n",
       "30  ENSBTAG00000016484     GO:0055037\n",
       "31  ENSBTAG00000012023     GO:0055114\n",
       "32  ENSBTAG00000014486     GO:0071805\n",
       "33  ENSBTAG00000014486     GO:0098655\n",
       "34  ENSBTAG00000012023     GO:0098869\n",
       "35  ENSBTAG00000016484     GO:0140326\n",
       "36  ENSBTAG00000014486     GO:1902600\n",
       "37  ENSBTAG00000014486     GO:1990573\n",
       "38  ENSBTAG00000012023  R-BTA-3299685\n",
       "39  ENSBTAG00000014486   R-BTA-936837\n",
       "40  ENSBTAG00000016484   R-BTA-936837"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table PUBLICATION:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract all record fields of a paper relative to an input pubmed_ID; \n",
      "3: to extract all record fields of a paper relative to an input doi; \n",
      "4: to extract all record fields of all papers relative to an input journal;\n",
      "5: to extract all record fields of all papers relative to an input author to search into field authors_name;\n",
      "6: to extract all record fields of all papers relative to an input publication year date;\n",
      "\n",
      "enter PUBLICATION table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pubmed_ID</th>\n",
       "      <th>doi</th>\n",
       "      <th>article_title</th>\n",
       "      <th>article_authors</th>\n",
       "      <th>article_journal</th>\n",
       "      <th>publication_year</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>29024632</td>\n",
       "      <td>10.1016/j.bbrc.2017.10.038</td>\n",
       "      <td>Quantitative proteome analysis of bovine mamma...</td>\n",
       "      <td>Xianrui Zheng, Chao Ning, Yichun Dong, Pengju ...</td>\n",
       "      <td>Biochemical and Biophysical Research Communica...</td>\n",
       "      <td>2017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   pubmed_ID                           doi  \\\n",
       "0   29024632   10.1016/j.bbrc.2017.10.038    \n",
       "\n",
       "                                       article_title  \\\n",
       "0  Quantitative proteome analysis of bovine mamma...   \n",
       "\n",
       "                                     article_authors  \\\n",
       "0  Xianrui Zheng, Chao Ning, Yichun Dong, Pengju ...   \n",
       "\n",
       "                                     article_journal  publication_year  \n",
       "0  Biochemical and Biophysical Research Communica...              2017  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table TECNIQUE:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract all record fields relative to an input tecnique; \n",
      "3: to extract all record fields relative to an input term to search into field tecnique_short_description;\n",
      "\n",
      "enter TECNIQUE table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tecnique</th>\n",
       "      <th>tecnique_short_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>Tandem Mass Tags label-based quantitative ana...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                tecnique                         tecnique_short_description\n",
       "0  TMT_LC-MS/MS_SDS-PAGE   Tandem Mass Tags label-based quantitative ana..."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table TAG:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract all record fields relative to an input keyword_tag; \n",
      "3: to extract all record fields relative to an input term to search into field keyword_tags; \n",
      "4: to extract all record fields relative to an input term to search into field tags_short_description;\n",
      "\n",
      "enter TAG table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>keyword_tags</th>\n",
       "      <th>tags_short_description</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>glutathione</td>\n",
       "      <td>genes related to glutathione metabolism and an...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GWAS</td>\n",
       "      <td>Genome-Wide Association Studies</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>holstein</td>\n",
       "      <td>holstein cattle breed</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>late lactation</td>\n",
       "      <td>the ultimate phase of lactation (biopsies mamm...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mammary gland</td>\n",
       "      <td>biopsies of mammary gland tissues at different...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>mastitis-free</td>\n",
       "      <td>biopsies of mastitis free mammary gland tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>peak lactation</td>\n",
       "      <td>the peak phase of lactation (biopsies mammary ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     keyword_tags                             tags_short_description\n",
       "0     glutathione  genes related to glutathione metabolism and an...\n",
       "1            GWAS                    Genome-Wide Association Studies\n",
       "2        holstein                              holstein cattle breed\n",
       "3  late lactation  the ultimate phase of lactation (biopsies mamm...\n",
       "4   mammary gland  biopsies of mammary gland tissues at different...\n",
       "5   mastitis-free    biopsies of mastitis free mammary gland tissues\n",
       "6  peak lactation  the peak phase of lactation (biopsies mammary ..."
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table VARIANT:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract all record fields relative to an input variant_name;\n",
      "3: to extract all record fields of all variants relative to an input ensembl_gene_ID;\n",
      "4: to extract all record fields of all variant relative to an input gene_name of an input species;\n",
      "5: to extract all record fields of all variants and the respective ensembl_gene_ID and gene_name relative to an input species; \n",
      "6: to extract all record fields of all variants and the respective ensembl_gene_ID and gene_name relative to an input species and an input chromosome;\n",
      "\n",
      "enter VARIANT table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant_name</th>\n",
       "      <th>variant_type</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>reference_allele</th>\n",
       "      <th>alternative_allele</th>\n",
       "      <th>rs_ID</th>\n",
       "      <th>species</th>\n",
       "      <th>refseq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>esonic</td>\n",
       "      <td>1</td>\n",
       "      <td>3824700</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs0000000</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>intronic</td>\n",
       "      <td>7</td>\n",
       "      <td>62284800</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>rs1234567</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>missens_intronic</td>\n",
       "      <td>7</td>\n",
       "      <td>62284999</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9876543</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              variant_name      variant_type  chromosome  position  \\\n",
       "0   BosTau11_1:3824700_C|A            esonic           1   3824700   \n",
       "1  BosTau11_7:62284800_A|G          intronic           7  62284800   \n",
       "2  BosTau11_7:62284999_A|T  missens_intronic           7  62284999   \n",
       "\n",
       "  reference_allele alternative_allele      rs_ID     species      refseq  \n",
       "0                C                  A  rs0000000  Bos taurus  ARS-UCD1.2  \n",
       "1                A                  G  rs1234567  Bos taurus  ARS-UCD1.2  \n",
       "2                A                  T  rs9876543  Bos taurus  ARS-UCD1.2  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n",
      "\n",
      "queires selection menu to export data from table PUB_GEN_VAR_TEC_TAG:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract pubmed_ID,doi,article_title of all publications related to an input ensembl_gene_ID; \n",
      "3: to extract pubmed_ID,doi,article_title of all publications related to an input gene_name; \n",
      "4: to extract ensembl_gene_ID and gene_name of all genes related to an input pubmed_ID; \n",
      "5: to extract ensembl_gene_ID and gene_name of all genes related to an input doi; \n",
      "6: to extract pubmed_ID,doi,article_title and ensembl_gene_ID and gene_name of all publications and genes related to an input keyword_tag; \n",
      "7: to extract ensembl_gene_ID and gene_name of all genes related to an input variant_name; \n",
      "8: to extract all tecnique informations of all tecniques related to an input pubmed_ID; \n",
      "9: to extract all database_ID informations of all database_IDs (gene, pathway, protein, enzyme...etc.) related to an input pubmed_ID; \n",
      "10: to extract pubmed_ID,doi,article_title,ensembl_gene_ID and gene_name of all publications and genes related to an input database_ID (database code of gene, pathway, protein, enzyme...etc.); \n",
      "11: to extract pubmed_ID,doi,article_title,ensembl_gene_ID and gene_name of all publications and genes related to an input compound name to search into database_object_name; \n",
      "12: to extract all record fields and gene_name related to an input pubmed_ID; \n",
      "13: to extract all record fields and gene_name related to an input doi; \n",
      "14: to extract pubmed_ID,doi,article_title and tecnique_name and keyword_tag related to an input ensembl_gene_ID; \n",
      "15: to extract pubmed_ID,doi,article_title and tecnique_name and keyword_tag related to an input gene_name;\n",
      "\n",
      "enter PUB_GEN_VAR_TEC_TAG table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>integer_progressive_ID</th>\n",
       "      <th>pubmed_ID</th>\n",
       "      <th>ensembl_gene_ID</th>\n",
       "      <th>variant_name</th>\n",
       "      <th>tecnique</th>\n",
       "      <th>keyword_tags</th>\n",
       "      <th>relationship_note</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>glutathione</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>glutathione</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>glutathione</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>glutathione</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000001842</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>glutathione</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>holstein</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>holstein</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>holstein</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>holstein</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000001842</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>holstein</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>mammary gland</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>mammary gland</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>mammary gland</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>mammary gland</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000001842</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>mammary gland</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>Mastitis-free</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>Mastitis-free</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>Mastitis-free</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>Mastitis-free</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000001842</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>Mastitis-free</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>peak lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>peak lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>peak lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>peak lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000001842</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>peak lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>late lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>late lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>late lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>late lactation</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284800_A|G</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000043553</td>\n",
       "      <td>BosTau11_7:62284999_A|T</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000018854</td>\n",
       "      <td>BosTau11_1:3824700_C|A</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000013980</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>29024632</td>\n",
       "      <td>ENSBTAG00000001842</td>\n",
       "      <td>None</td>\n",
       "      <td>TMT_LC-MS/MS_SDS-PAGE</td>\n",
       "      <td>GWAS</td>\n",
       "      <td>proteins identified by proteome tecnique pipel...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    integer_progressive_ID  pubmed_ID     ensembl_gene_ID  \\\n",
       "0                        1   29024632  ENSBTAG00000043553   \n",
       "1                        2   29024632  ENSBTAG00000043553   \n",
       "2                        3   29024632  ENSBTAG00000018854   \n",
       "3                        4   29024632  ENSBTAG00000013980   \n",
       "4                        5   29024632  ENSBTAG00000001842   \n",
       "5                        6   29024632  ENSBTAG00000043553   \n",
       "6                        7   29024632  ENSBTAG00000043553   \n",
       "7                        8   29024632  ENSBTAG00000018854   \n",
       "8                        9   29024632  ENSBTAG00000013980   \n",
       "9                       10   29024632  ENSBTAG00000001842   \n",
       "10                      11   29024632  ENSBTAG00000043553   \n",
       "11                      12   29024632  ENSBTAG00000043553   \n",
       "12                      13   29024632  ENSBTAG00000018854   \n",
       "13                      14   29024632  ENSBTAG00000013980   \n",
       "14                      15   29024632  ENSBTAG00000001842   \n",
       "15                      16   29024632  ENSBTAG00000043553   \n",
       "16                      17   29024632  ENSBTAG00000043553   \n",
       "17                      18   29024632  ENSBTAG00000018854   \n",
       "18                      19   29024632  ENSBTAG00000013980   \n",
       "19                      20   29024632  ENSBTAG00000001842   \n",
       "20                      21   29024632  ENSBTAG00000043553   \n",
       "21                      22   29024632  ENSBTAG00000043553   \n",
       "22                      23   29024632  ENSBTAG00000018854   \n",
       "23                      24   29024632  ENSBTAG00000013980   \n",
       "24                      25   29024632  ENSBTAG00000001842   \n",
       "25                      26   29024632  ENSBTAG00000043553   \n",
       "26                      27   29024632  ENSBTAG00000043553   \n",
       "27                      28   29024632  ENSBTAG00000018854   \n",
       "28                      29   29024632  ENSBTAG00000013980   \n",
       "29                      30   29024632  ENSBTAG00000043553   \n",
       "30                      31   29024632  ENSBTAG00000043553   \n",
       "31                      32   29024632  ENSBTAG00000018854   \n",
       "32                      33   29024632  ENSBTAG00000013980   \n",
       "33                      34   29024632  ENSBTAG00000001842   \n",
       "\n",
       "               variant_name               tecnique    keyword_tags  \\\n",
       "0   BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE     glutathione   \n",
       "1   BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE     glutathione   \n",
       "2    BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE     glutathione   \n",
       "3                      None  TMT_LC-MS/MS_SDS-PAGE     glutathione   \n",
       "4                      None  TMT_LC-MS/MS_SDS-PAGE     glutathione   \n",
       "5   BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE        holstein   \n",
       "6   BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE        holstein   \n",
       "7    BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE        holstein   \n",
       "8                      None  TMT_LC-MS/MS_SDS-PAGE        holstein   \n",
       "9                      None  TMT_LC-MS/MS_SDS-PAGE        holstein   \n",
       "10  BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE   mammary gland   \n",
       "11  BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE   mammary gland   \n",
       "12   BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE   mammary gland   \n",
       "13                     None  TMT_LC-MS/MS_SDS-PAGE   mammary gland   \n",
       "14                     None  TMT_LC-MS/MS_SDS-PAGE   mammary gland   \n",
       "15  BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE   Mastitis-free   \n",
       "16  BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE   Mastitis-free   \n",
       "17   BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE   Mastitis-free   \n",
       "18                     None  TMT_LC-MS/MS_SDS-PAGE   Mastitis-free   \n",
       "19                     None  TMT_LC-MS/MS_SDS-PAGE   Mastitis-free   \n",
       "20  BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE  peak lactation   \n",
       "21  BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE  peak lactation   \n",
       "22   BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE  peak lactation   \n",
       "23                     None  TMT_LC-MS/MS_SDS-PAGE  peak lactation   \n",
       "24                     None  TMT_LC-MS/MS_SDS-PAGE  peak lactation   \n",
       "25  BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE  late lactation   \n",
       "26  BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE  late lactation   \n",
       "27   BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE  late lactation   \n",
       "28                     None  TMT_LC-MS/MS_SDS-PAGE  late lactation   \n",
       "29  BosTau11_7:62284800_A|G  TMT_LC-MS/MS_SDS-PAGE            GWAS   \n",
       "30  BosTau11_7:62284999_A|T  TMT_LC-MS/MS_SDS-PAGE            GWAS   \n",
       "31   BosTau11_1:3824700_C|A  TMT_LC-MS/MS_SDS-PAGE            GWAS   \n",
       "32                     None  TMT_LC-MS/MS_SDS-PAGE            GWAS   \n",
       "33                     None  TMT_LC-MS/MS_SDS-PAGE            GWAS   \n",
       "\n",
       "                                    relationship_note  \n",
       "0   proteins identified by proteome tecnique pipel...  \n",
       "1   proteins identified by proteome tecnique pipel...  \n",
       "2   proteins identified by proteome tecnique pipel...  \n",
       "3   proteins identified by proteome tecnique pipel...  \n",
       "4   proteins identified by proteome tecnique pipel...  \n",
       "5   proteins identified by proteome tecnique pipel...  \n",
       "6   proteins identified by proteome tecnique pipel...  \n",
       "7   proteins identified by proteome tecnique pipel...  \n",
       "8   proteins identified by proteome tecnique pipel...  \n",
       "9   proteins identified by proteome tecnique pipel...  \n",
       "10  proteins identified by proteome tecnique pipel...  \n",
       "11  proteins identified by proteome tecnique pipel...  \n",
       "12  proteins identified by proteome tecnique pipel...  \n",
       "13  proteins identified by proteome tecnique pipel...  \n",
       "14  proteins identified by proteome tecnique pipel...  \n",
       "15  proteins identified by proteome tecnique pipel...  \n",
       "16  proteins identified by proteome tecnique pipel...  \n",
       "17  proteins identified by proteome tecnique pipel...  \n",
       "18  proteins identified by proteome tecnique pipel...  \n",
       "19  proteins identified by proteome tecnique pipel...  \n",
       "20  proteins identified by proteome tecnique pipel...  \n",
       "21  proteins identified by proteome tecnique pipel...  \n",
       "22  proteins identified by proteome tecnique pipel...  \n",
       "23  proteins identified by proteome tecnique pipel...  \n",
       "24  proteins identified by proteome tecnique pipel...  \n",
       "25  proteins identified by proteome tecnique pipel...  \n",
       "26  proteins identified by proteome tecnique pipel...  \n",
       "27  proteins identified by proteome tecnique pipel...  \n",
       "28  proteins identified by proteome tecnique pipel...  \n",
       "29  proteins identified by proteome tecnique pipel...  \n",
       "30  proteins identified by proteome tecnique pipel...  \n",
       "31  proteins identified by proteome tecnique pipel...  \n",
       "32  proteins identified by proteome tecnique pipel...  \n",
       "33  proteins identified by proteome tecnique pipel...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n"
     ]
    }
   ],
   "source": [
    "'''\n",
    "pignotti.danilo@gmail.com\n",
    "emanuele.tufarini@live.com\n",
    "last modification 12/2020\n",
    "'''\n",
    "\n",
    "import os, sys\n",
    "directory = os.getcwd()\n",
    "# import functions from the functions folder\n",
    "library_directory = directory + \"/FUNCTIONS/\"\n",
    "sys.path.append(library_directory)\n",
    "# import database export function\n",
    "from MIQUALAT_data_export import *\n",
    "\n",
    "# database export \n",
    "miqualat_db_data_exporter()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
