{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# run to export data from the database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "enter your username: root\n",
      "enter your password: ········\n",
      "\n",
      "tables selection menu from which to export data:\n",
      "\n",
      "enter 1 to export data from database MIQUALAT following table: GENE;\n",
      "enter 2 to export data from database MIQUALAT following table: DB;\n",
      "enter 3 to export data from database MIQUALAT following table: GEN_DB;\n",
      "enter 4 to export data from database MIQUALAT following table: PUBLICATION;\n",
      "enter 5 to export data from database MIQUALAT following table: TECNIQUE;\n",
      "enter 6 to export data from database MIQUALAT following table: TAG;\n",
      "enter 7 to export data from database MIQUALAT following table: VARIANT;\n",
      "enter 8 to export data from database MIQUALAT following table: PUB_GEN_VAR_TEC_TAG;\n",
      "\n",
      "enter tables selection numbers (i.e. 7 or 1,2,3): 1\n",
      "\n",
      "queires selection menu to export data from table GENE:\n",
      " \n",
      "1: to extract all table records; \n",
      "2: to extract specific record relative to an input ensembl ID; \n",
      "3: to extract specific record relative to an input gene_name; \n",
      "4: to extract all the species and relative refseq version; \n",
      "5: to count genes total numbers sorted by species; \n",
      "6: to extract all genes from an input species; \n",
      "7: to extract all genes located on an input chromosome for an input species; \n",
      "8: to extract all genes located between start and end input coordinates on an input chromosome for an input species;\n",
      "\n",
      "enter GENE table queries selection numbers (i.e. 4 or 1,2,3): 1\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_ID</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>gene_short_description</th>\n",
       "      <th>refseq</th>\n",
       "      <th>species</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start_coordinate</th>\n",
       "      <th>end_coordinate</th>\n",
       "      <th>strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSBTAG00000000005</td>\n",
       "      <td>GRK3</td>\n",
       "      <td>G protein-coupled receptor kinase 3</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>17</td>\n",
       "      <td>65389743</td>\n",
       "      <td>65505336</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSBTAG00000000008</td>\n",
       "      <td>KCNJ1</td>\n",
       "      <td>potassium inwardly rectifying channel subfamil...</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>29</td>\n",
       "      <td>32214439</td>\n",
       "      <td>32244810</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSBTAG00000000009</td>\n",
       "      <td>FOXF1</td>\n",
       "      <td>forkhead box F1</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>18</td>\n",
       "      <td>12338037</td>\n",
       "      <td>12342272</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSBTAG00000000010</td>\n",
       "      <td>UBL7</td>\n",
       "      <td>ubiquitin like 7</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>21</td>\n",
       "      <td>34209956</td>\n",
       "      <td>34223394</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSBTAG00000000011</td>\n",
       "      <td>TDH</td>\n",
       "      <td>L-threonine dehydrogenase</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>8</td>\n",
       "      <td>7950815</td>\n",
       "      <td>7971600</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27265</th>\n",
       "      <td>ENSBTAG00000055312</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>2</td>\n",
       "      <td>96551552</td>\n",
       "      <td>96557130</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27266</th>\n",
       "      <td>ENSBTAG00000055313</td>\n",
       "      <td>bta-mir-12025</td>\n",
       "      <td>bta-mir-12025</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>21</td>\n",
       "      <td>65169462</td>\n",
       "      <td>65169520</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27267</th>\n",
       "      <td>ENSBTAG00000055314</td>\n",
       "      <td>OR3A1</td>\n",
       "      <td>olfactory receptor, family 3, subfamily A, mem...</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>19</td>\n",
       "      <td>24021135</td>\n",
       "      <td>24051219</td>\n",
       "      <td>-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27268</th>\n",
       "      <td>ENSBTAG00000055315</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>8</td>\n",
       "      <td>102574538</td>\n",
       "      <td>102575815</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27269</th>\n",
       "      <td>ENSBTAG00000055316</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ARS-UCD1.2</td>\n",
       "      <td>bos taurus</td>\n",
       "      <td>2</td>\n",
       "      <td>94738416</td>\n",
       "      <td>94738887</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27270 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          ensembl_gene_ID      gene_name  \\\n",
       "0      ENSBTAG00000000005           GRK3   \n",
       "1      ENSBTAG00000000008          KCNJ1   \n",
       "2      ENSBTAG00000000009          FOXF1   \n",
       "3      ENSBTAG00000000010           UBL7   \n",
       "4      ENSBTAG00000000011            TDH   \n",
       "...                   ...            ...   \n",
       "27265  ENSBTAG00000055312           None   \n",
       "27266  ENSBTAG00000055313  bta-mir-12025   \n",
       "27267  ENSBTAG00000055314          OR3A1   \n",
       "27268  ENSBTAG00000055315           None   \n",
       "27269  ENSBTAG00000055316           None   \n",
       "\n",
       "                                  gene_short_description      refseq  \\\n",
       "0                   G protein-coupled receptor kinase 3   ARS-UCD1.2   \n",
       "1      potassium inwardly rectifying channel subfamil...  ARS-UCD1.2   \n",
       "2                                       forkhead box F1   ARS-UCD1.2   \n",
       "3                                      ubiquitin like 7   ARS-UCD1.2   \n",
       "4                             L-threonine dehydrogenase   ARS-UCD1.2   \n",
       "...                                                  ...         ...   \n",
       "27265                                               None  ARS-UCD1.2   \n",
       "27266                                     bta-mir-12025   ARS-UCD1.2   \n",
       "27267  olfactory receptor, family 3, subfamily A, mem...  ARS-UCD1.2   \n",
       "27268                                               None  ARS-UCD1.2   \n",
       "27269                                               None  ARS-UCD1.2   \n",
       "\n",
       "          species  chromosome  start_coordinate  end_coordinate  strand  \n",
       "0      bos taurus          17          65389743        65505336       1  \n",
       "1      bos taurus          29          32214439        32244810      -1  \n",
       "2      bos taurus          18          12338037        12342272       1  \n",
       "3      bos taurus          21          34209956        34223394       1  \n",
       "4      bos taurus           8           7950815         7971600      -1  \n",
       "...           ...         ...               ...             ...     ...  \n",
       "27265  bos taurus           2          96551552        96557130       1  \n",
       "27266  bos taurus          21          65169462        65169520       1  \n",
       "27267  bos taurus          19          24021135        24051219      -1  \n",
       "27268  bos taurus           8         102574538       102575815       1  \n",
       "27269  bos taurus           2          94738416        94738887       1  \n",
       "\n",
       "[27270 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "after displayed exportation example do you want to save results in a .csv file in OUTPUT folder? (i.e. yes or not): yes\n"
     ]
    }
   ],
   "source": [
    "'''\n",
    "pignotti.danilo@gmail.com\n",
    "emanuele.tufarini@live.com\n",
    "last modification 12/2020\n",
    "'''\n",
    "\n",
    "import os, sys\n",
    "directory = os.getcwd()\n",
    "# import functions from the functions folder\n",
    "library_directory = directory + \"/FUNCTIONS/\"\n",
    "sys.path.append(library_directory)\n",
    "# import database export function\n",
    "from MIQUALAT_data_export import *\n",
    "\n",
    "# database export \n",
    "miqualat_db_data_exporter(host=\"localhost\",db=\"MIQUALAT\",port=3306)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
